strapi/strapi
# enhancement: add back button fallback support

### What does it do?

Adds a fallback prop to the BackButton from `@strapi/admin/strapi-admin`.

This has been approved by Yannis [on slack (plus video for more context)](https://strapihq.slack.com/archives/C051PLC9CF7/p1729774440081529)

### Why is it needed?

I want to avoid the case where the back button is disabled, as that UX isn't great imo. It happens when you reload your browser window, or if you land on a page via a bookmark or a link someone sent you.

In a lot of cases, the page . For example, CM list view > CM edit view. It doesn't work everywhere, but this PR adds the API so we can tell where the back button should lead when no history is found. In a lot of cases it looks like this with react router:

```jsx
<BackButton fallback=".." />
```

Note that when relying on the fallback prop, it just adds to the history stack, it doesn't emulate the browser's back button. IMO this is an okay tradeoff.

### How to test it?

In each of these pages, reload the page before clicking on the back button. Check that you are redirect to the page that you expected (the "parent" page).

- cm > edit view for a collection type
- cm > edit view for a single type (should be disabled)
- cm > list view > configure the view
- settings > admin roles > create
- settings > admin roles > edit
- settings > admin users > edit
- settings > webhooks > create/edit
- settings > review workflows > workflow
- settings > users & permissions > role

- releases > release details page
